packages/dialog/src/Dialog.ts

```diff
@@ -47,7 +49,7 @@ function gatherAppliedIdsFromSlottedChildren(
         if (el.id) {
             ids.push(el.id);
         } else {
-            const id = idBase + `-${i}`;
+            const id = idBase + `-${i + appliedIds++}`;
```

Comment:
It might be simpler to understand to have just one, globally-unique source of truth for the IDs:

```suggestion
            const id = `${idBase}-${appliedIds++}`;
```

Reply:
Good call.